### PR TITLE
Ts node support + tests

### PR DESCRIPTION
**Pull request by chgeo**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,7 +39,18 @@
     "parallel": true
   },
   "jest": {
-    "testEnvironment": "node"
+    "testEnvironment": "node",
+    "transform": {
+      "\\.ts$": "ts-jest"
+    },
```
Comment (chgeo) [inline package.json:45]: @Kkoile this `transform` option finally did the trick

File: package.json [hunk 2]
```diff
@@ -36,7 +39,18 @@
...
+    "globals": {
+      "ts-jest": {
+        "diagnostics": {
+          "_comment": "see https://githubmemory.com/repo/kulshekhar/ts-jest/issues/2722",
+          "ignoreCodes": [151001]
+        }
```
Comment (chgeo) [inline package.json:51]: This is to suppress a useless warning of ts compiler

File: package.json (hunk 1 above)
Comment (chgeo) [reply package.json]: replaced by `preset: 'ts-jest'`

**Comment by Kkoile** [comment]
I have two concerns: 
1) Because you put the preset: ts-jest in the root jest.config, it applies for all tests. Even the non typescript tests. Do they still work? Is it wanted?
2) I had to manually set the environment variabel `CDS_TYPESCRIPT=true` in the tests for cds to recognize that it should load typescript files now. It had to be done, because it is not possible to programatically detect, if a process has started with ts-jest preset. I don't see this variable anywhere in the code now. Has something changed, that it is no longer required to set the environment variable? Tests seem to go through...

**Comment by Kkoile** [comment]
Oh, regarding my second point, I've just seen that the environment variable is actually set: https://github.com/SAP-samples/cloud-cap-samples/pull/248/files#diff-2d29ae20792820e80f8924914e7010e5e25076584cff77b50001f7353d7860a9R1
I wonder though, whether it will break other tests in the future? Probably it would be better to include this
```typescript
afterAll(() => {
  delete process.env.CDS_TYPESCRIPT;
});
```
Or remember which was the original value before assigning it `true` and then set it to the original value in the after handler

**Comment by tsteckenborn** [comment]
> I have two concerns:
> 
> 1. Because you put the preset: ts-jest in the root jest.config, it applies for all tests. Even the non typescript tests. Do they still work? Is it wanted?

The preset says quite explicitly:

> TypeScript files (.ts, .tsx) will be transformed by ts-jest to CommonJS syntax, leaving JavaScript files (.js, jsx) as-is.

So I would expect it to do exactly the same as before (but including tsx for transformation).

**Comment by chgeo** [comment]
> So I would expect it to do exactly the same as before (but including tsx for transformation).

Yes, that's also my assumption.  If it doesn't harm or slow down the js tests too much, then fine with me.

**Comment by chgeo** [comment]
> I wonder though, whether it will break other tests in the future? Probably it would be better to include this
> 
> ```ts
> afterAll(() => {
>   delete process.env.CDS_TYPESCRIPT;
> });
> ```
> 

Done

**Comment by tsteckenborn** [comment]
Otherwise, the question is whether there should be a separate sample repository for TypeScript (as it would need to contain ts-config and maybe other configurations in the long run) to separate the necessary configurations of JavaScript and TypeScript for newcomers.

It will be hard to distinguish what is not to compromise the JS parts and what is (really) necessary if I go the TS way. Deleting the variables afterward, for example, shouldn't be required if I decide to go complete TypeScript, right? 

Therefore, it could not be obvious. Since the environment variable is not even described in capire or the changelog, it's a bit of a search for users who want to test the TypeScript files anyways.

**Comment by chgeo** [comment]
@Kkoile we seem to have issues w/ your CLA.  Can you check?  Could be that you used a wrong email address not connected to your current GH account.

**Comment by chgeo** [comment]
> Otherwise, the question is whether there should be a separate sample repository for TypeScript 

That's a constant trade-off in how many repos to cut the samples, given that there is maintenance overhead of each new one.  That was the reason for the current mono-repo.
I do the see the potential for TS/JS confusion, though.

> Since the environment variable is not even described in capire or the changelog

[It is](https://cap.cloud.sap/docs/get-started/using-typescript#testing-with-ts-jest) (with the wrong value though, about to fix this).

**Comment by chgeo** [comment]
@Kkoile I have rewritten your commits with my email address if you don't mind.  This way, we get over the CLA barrier.

**Comment by chgeo** [comment]
All TS related stuff is now isolated in the `hello` folder, so that it's cleared where TS land is.

**Comment by Kkoile** [comment]
Sorry, I'm always lacking a bit behind with my private github account. But sinced you merge the pull request, I guess everything is clear now and there's nothing more to do?

**Comment by chgeo** [comment]
Right, all is settled :)

**Comment by chgeo** [comment]
@tsteckenborn you can reach out to me (christian.georgi@sap.com) and maybe nils.hirsekorn@sap.com (@Kkoile), who is an active TS user here at SAP.

(@iwonahahn FYI)

**Pull request merged**